packages/toolkit/src/amazonqGumby/activation.ts

```diff
@@ -54,29 +83,28 @@ export async function activate(context: ExtContext) {
                 'markdown.showPreview',
                 vscode.Uri.file(transformByQState.getPlanFilePath())
             )
-        })
+        }),
+
+        workspaceChangeEvent
     )
 
-    // Register an activation event listener to determine when the IDE opens, closes or users
-    // select to open a new workspace
-    vscode.workspace.onDidChangeWorkspaceFolders(event => {
-        // Register when the IDE is closed
-        if (event.added.length === 0 && event.removed.length === 0) {
-            // Only fire closed during running/active job status
-            if (transformByQState.isRunning()) {
-                telemetry.codeTransform_jobIsClosedDuringIdeRun.emit({
-                    codeTransformJobId: transformByQState.getJobId(),
-                    codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-                    codeTransformStatus: transformByQState.getStatus(),
-                })
-            }
-        } else {
-            // Register when the workspace is changed to a new project, or IDE is opened
-            telemetry.codeTransform_jobIsResumedAfterIdeClose.emit({
-                codeTransformJobId: transformByQState.getJobId(),
-                codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-                codeTransformStatus: transformByQState.getStatus(),
-            })
-        }
-    })
+    // Try to validate project silently
+    try {
+        const openProjects = await getOpenProjects()
+        const validProjects = await validateOpenProjects(openProjects)
+        const firstKey = validProjects.keys().next().value
+        const firstModuleEntry = validProjects.get(firstKey)
+        telemetry.codeTransform_projectDetails.emit({
+            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+            codeTransformLocalJavaVersion:
+                (JDKToTelemetryValue(firstModuleEntry) as CodeTransformJavaSourceVersionsAllowed) || undefined,
+        })
+    } catch (err: any) {
+        telemetry.codeTransform_projectDetails.emit({
```

Comment:
This will always be emitted for every user. Why?

Reply:
This is a P0 requirement from leadership.

Reply:
> Maybe we can reduce this to just be Codewhisperer users, or eligible users of Gumby.

Does the `transformByQState` global have something that tells you "this user has touched gumby"? Most users don't use gumby, so it is not correct to emit gumby metrics for them (that makes telemetry very noise, which makes dashboards harder to reason about).